Drivers/IWDG_driver/Src/IndependentWatchdog.cpp

```diff
@@ -7,39 +7,54 @@
  *
  */
 IWDG_t::IWDG_t(IWDG_HandleTypeDef *watchdog) {
+	this->watchdog = watchdog;
 	IWDG_InitTypeDef initDef = watchdog->Init;
 	this->iwdg_prescaler = initDef.Prescaler;
 	this->iwdg_reload = initDef.Reload;
 }
 
+
 /**
- * @brief Initalize the driver using custom IWDG_Prescaler and realod values.
+ * @brief Reset the reload and counter values inside a watchdog
+ * using the member variables of IWDG_t
  *
  */
-IWDG_t::IWDG_t(uint8_t iwdg_prescaler, uint16_t iwdg_reload) {
-  // assert(IS_IWDG_PRESCALER(iwdg_prescaler)) Should this be added since it is needed when reinitialzing the watchdog
-  // assert(IS_IWDG_RELOAD(iwdg_reload))
-	this->iwdg_prescaler = iwdg_prescaler;
-	this->iwdg_reload = iwdg_reload;
+bool IWDG_t::SetWatchdogParameters(uint8_t prescaler, uint16_t reload) {
+	this->iwdg_prescaler = prescaler;
+	this->iwdg_reload = reload;
+
+	if (!IS_IWDG_PRESCALER(prescaler) || !IS_IWDG_RELOAD(reload)) return false;
+	watchdog->Init.Reload = reload;
+	watchdog->Init.Prescaler = prescaler;
+	HAL_StatusTypeDef refresh = HAL_IWDG_Init(watchdog);
+	return (refresh==HAL_OK);
 }
 
+
 /**
- * @brief Reset the reload and counter values inside a watchdog
- * using the member variables of IWDG_t
+ * @brief Refreshes the watchdog that is a member variable of the class
  *
  */
-HAL_StatusTypeDef IWDG_t::InitWatchdog(IWDG_HandleTypeDef *watchdog) {
-	watchdog->Init.Reload = this->iwdg_reload;
-	watchdog->Init.Prescaler = this->iwdg_prescaler;
-	HAL_StatusTypeDef refresh = HAL_IWDG_Init(watchdog);
-	return refresh;
+
+bool IWDG_t::RefreshWatchdog() {
+	if (!this->watchdog) {
+		return false;
+	}
+	return (HAL_IWDG_Refresh(watchdog) == HAL_OK);
```

Comment:
Do we need to be concerned about thread safety here?
Will multiple threads need to pet the watchdog, and if so is this operation thread safe?